virttest/utils_conn.py

```diff
@@ -1941,41 +1989,68 @@ def conn_setup(self):
 
         # restart libvirtd service on server
         if restart_libvirtd == "yes":
             try:
+                if not self.run_on_remote:
+                    current_session = client_session
+                    runner = process.run
+                else:
+                    current_session = remote.wait_for_login('ssh',
+                                                            server_ip,
+                                                            '22',
+                                                            server_user,
+                                                            server_pwd,
+                                                            r"[\#\$]\s*$")
+                    remote_session = remote.wait_for_login('ssh',
+                                                           server_ip,
+                                                           '22',
+                                                           server_user,
+                                                           server_pwd,
+                                                           r"[\#\$]\s*$")
+                    remote_runner = remote_old.RemoteRunner(
+                        session=remote_session)
+                    runner = remote_runner.run
                 libvirtd_service = utils_libvirtd.Libvirtd(
-                    session=client_session)
+                    session=current_session)
                 if self.libvirt_ver:
                     # For traditional mode, need mask libvirtd*.socket
                     if traditional_mode == 'yes':
-                        process.run("systemctl stop libvirtd.socket", shell=True)
-                        process.run("systemctl stop libvirtd-admin.socket", shell=True)
-                        process.run("systemctl stop libvirtd-ro.socket", shell=True)
-                        process.run("systemctl stop libvirtd-tcp.socket", shell=True)
-                        process.run("systemctl stop libvirtd-tls.socket", shell=True)
+                        runner("systemctl stop libvirtd.socket")
+                        runner("systemctl stop libvirtd-admin.socket")
+                        runner("systemctl stop libvirtd-ro.socket")
+                        runner("systemctl stop libvirtd-tcp.socket")
+                        runner("systemctl stop libvirtd-tls.socket")
                         libvirtd_service.stop()
-                        process.run("systemctl mask libvirtd.socket", shell=True)
-                        process.run("systemctl mask libvirtd-admin.socket", shell=True)
-                        process.run("systemctl mask libvirtd-ro.socket", shell=True)
-                        process.run("systemctl mask libvirtd-tcp.socket", shell=True)
-                        process.run("systemctl mask libvirtd-tls.socket", shell=True)
-                        process.run("systemctl daemon-reload", shell=True)
+                        runner("systemctl mask libvirtd.socket")
```

Comment:
Is there any potential risks to remove 'shell=True' for process.run?

Reply:
I didn't found any. The shell = True will just create a subshell during the execution and it is not supported by the remote runner, so I didn't found a reason to have it here. Are you aware of any possible issue?

Reply:
I remember process.run might get certain unexpected error without shell=True sometimes. But if you have run all the related test cases, I am fine with it so far.